### PR TITLE
Add ERC20 interface and Token implementation w/ tests

### PR DESCRIPTION
**Pull request by lknix**

### CODE REVIEW

File: contracts/tokens/Token.sol [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+ * @dev Standard ERC20 token. This contract follows the
+ * implementation at https://github.com/ethereum/EIPs/blob/master/EIPS/eip-20.md
+ */
+contract Token {
+  using SafeMath for uint256;
+
+  /**
+   * Token name.
+   */
+  string tokenName;
```
Comment (xpepermint) [inline contracts/tokens/Token.sol]: add visibility

File: contracts/tokens/Token.sol [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+
+  /**
+   * Token symbol.
+   */
+  string tokenSymbol;
```
Comment (xpepermint) [inline contracts/tokens/Token.sol]: add visibility

File: contracts/tokens/Token.sol [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+
+  /**
+   * Number of decimals.
+   */
+  uint8 tokenDecimals;
```
Comment (xpepermint) [inline contracts/tokens/Token.sol]: add visibility

File: contracts/tokens/Token.sol [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+
+  /**
+   * Total supply of tokens.
+   */
+  uint256 tokenTotalSupply;
```
Comment (xpepermint) [inline contracts/tokens/Token.sol]: add visibility

File: contracts/tokens/Token.sol [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+
+  /**
+   * Token allowance mapping.
+   */
+  mapping (address => mapping (address => uint256)) internal allowed;
+
+  /**
+   * @dev Trigger when tokens are transferred, including zero value transfers.
+   */
+  event Transfer(address indexed _from, address indexed _to, uint256 _value);
```
Comment (xpepermint) [inline contracts/tokens/Token.sol]: follow 0xcert code style

File: contracts/tokens/Token.sol [hunk 6]
```diff
@@ -0,0 +1,183 @@
...
+
+  /**
+   * @dev Trigger on any successful call to approve(address _spender, uint256 _value).
+   */
+  event Approval(address indexed _owner, address indexed _spender, uint256 _value);
```
Comment (xpepermint) [inline contracts/tokens/Token.sol]: follow 0xcert code style

File: test/tokenks/Token.test.js [hunk 7]
```diff
@@ -0,0 +1,116 @@
+const Token = artifacts.require('TokenMock');
+const assertRevert = require('../../node_modules/@0xcert/ethereum-utils/test/helpers/assertRevert');
+
+contract('erc/Token', (accounts) => {
+  let token;
+  let owner = accounts[0];
+  let recipient = accounts[1];
```
Comment (xpepermint) [inline test/tokenks/Token.test.js]: user `const` instead of `let` if possible

File: contracts/tokens/Token.sol (hunk 6 above)
Comment (lknix) [reply contracts/tokens/Token.sol]: b21430d

File: test/tokenks/Token.test.js (hunk 7 above)
Comment (lknix) [reply test/tokenks/Token.test.js]: 2c8124e

File: contracts/tokens/Token.sol [hunk 8]
```diff
@@ -0,0 +1,183 @@
...
+  {
+    _decimals = tokenDecimals;
+  }
+
+  /**
+   * @dev Returns the total token supply.
+   */
+  function totalSupply()
+    external
+    constant
```
[the remark this answers is not in the document]
Comment (lknix) [reply contracts/tokens/Token.sol]: f92b4ed

**Pull request merged**